evennia/utils/evmenu.py

```diff
@@ -1138,6 +1325,272 @@ def node_formatter(self, nodetext, optionstext):
         separator2 = "\n" + sep * total_width + "\n\n" if total_width else ""
         return separator1 + "|n" + nodetext + "|n" + separator2 + "|n" + optionstext
 
+    def options_format_bars(self, callback, field, **nodedata):
```

Comment:
As said, I disagree about the use of distinct methods for this, but think it's way too many special cases for the dev to remember. One standard-named `formatter` callable with optional kwargs to specify different formatting options is a lot less mental overhead. If wanting to keep things tidy, you could have the main default options formatter check the kwargs and call different sub-functions depending on what options are given.

Reply:
Also, what's with the unused `callback` argument everywhere?

Reply:
I don't really see how it is less mental overhead for a developer to say that they want to use the ``no_bars`` format which calls ''options_format_no_bars`` over saying they want to use the ``no_bars`` format which calls ``formatter`` and then decides what part of the code to use within that method. Either way the developer is typing the exact same things.

At present the developers only really need to worry about 5 options, three for the node itself (default, invert, and suppress), and 2 for all the sub-entries to the node (bars and no_bars). That's why a modification is appended to the call name (so that ``bars`` for help calls a different formatter than ``bars`` for options)

The advantage to using the individual methods is that if someone wants to create a new formatter such as a ``box`` formatter (one that not only creates bars on the top and bottom but also creates vertical bar glyphs on the sides) they only need to create a new formatter function for their specific override (or two or three functions if they want boxes around text, options, and/or the assembled node itself). 

They do not need to go and copy the existing ``formatter`` function into the override and place their code inside which increases the possibility of breaking something. Also, by making the individual formatters their own cases it makes code sharing easier. Rather than having to cut formats from different sources and paste them into the ``formatter`` method they can simply be pasted `as is` into the override (with the possible exception that a name might need to be changed if there's naming collision between the formats).

As for the unusued ``callback`` argument, I'm using the ``_safe_call`` function you wrote to call the individual formatters. This means that when I add ``text_format_box`` to my override I don't need to go in somewhere else and add ``if field['formatter'] == 'box':``. 

From what I was able to deduce it seems like _safe_call demands that a variable be there in that position for callback to previous nodes. I was going to talk about that in the tutorial I plan to write for EvMenu.

Reply:
I can possibly accept your argument on the basis of  an easier overload. If `formatter=` key allows both key, callable and kwargs to be sent as discussed, I suppose it'd be ok to provide some default methods to call even if they are separate methods.

Are you referring to the `EvMenu._safe_call` method? It doesn't require its input to have a callback? It only requires the callable to have a at minimum a `caller` as its first argument, with  an optional range of other possible args/kwargs. It's meant to execute nodes and node-like callables with variable arguments. 
Requiring useless boiler plate to appease `_save_call` goes against the purpose of that method though  - it's meant to give the dev more freedom for how their callables may look, not the other way around. For the formatters you have no need for the `caller` argument so if you want to use the `_safe_call` method you should expand on _safe_call to also handle the possible arguments to a formatter.